mava/components/jax/building/data_server.py

```diff
@@ -107,6 +112,21 @@ def name() -> str:
         """Component type name, e.g. 'dataset' or 'executor'."""
         return "data_server"
 
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        BaseSystemInit required to set up builder.store.agent_net_keys
+        and for config network_sampling_setup.
+        EnvironmentSpec required to set up builder.store.environment_spec
+        and builder.store.extras_spec.
+        AdderSignature required to set up builder.store.adder_signature_fn.
```

Comment:
General comment also applicable elsewhere, but I really appreciate the detailed descriptions of which components are required.

Reply:
Agreed :smile: